cachelib/allocator/CacheAllocator-inl.h

```diff
@@ -2536,7 +2636,17 @@ bool CacheAllocator<CacheTrait>::moveForSlabRelease(
 
     if (!newItemHdl) {
       // try to allocate again if it previously wasn't successful
-      newItemHdl = allocateNewItemForOldItem(oldItem);
+      if (chainedItem) {
+        parentItem = &oldItem.asChainedItem().getParentItem(compressor_);
+        XDCHECK(parentItem->isMoving());
+        XDCHECK(oldItem.isChainedItem() && oldItem.getRefCount() == 1);
+        XDCHECK_EQ(0, parentItem->getRefCount());
+        newItemHdl =
+            allocateChainedItemInternal(*parentItem, oldItem.getSize());
```

Comment:
I think we need to call `allocateNewItemForOldItem` function for chained items as well. And modify `allocateNewItemForOldItem` function to work correctly with chained items.